### PR TITLE
fix(core): ensure needsReload flag works

### PR DESCRIPTION
**Pull request by eysi09**
**What this PR does / why we need it**:

That's a function, not a boolean we were evaluating so it always returned true. This fixes that.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by TimBeyer — APPROVED** [review]

File: core/src/server/server.ts [hunk 1]
```diff
@@ -892,7 +892,7 @@ export class GardenServer extends EventEmitter {
       } catch (error) {
         errors.push(toGardenError(error))
       } finally {
-        loadConfigLog.success(`Loading config failed with error: ${errors[0].message}`)
+        loadConfigLog.error(`Loading config failed with error: ${errors[0].message}`)
```
[the remark this answers is not in the document]
Comment (stefreak) [reply core/src/server/server.ts:895]: Is it possible that this line of code should be above the line 894?

**Review by stefreak — CHANGES_REQUESTED** [review]

File: core/src/server/server.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimBeyer) [reply core/src/server/server.ts:895]: Good catch, revoking my approval 😅

File: core/src/server/instance-manager.ts [hunk 2]
```diff
@@ -236,7 +236,7 @@ export class GardenInstanceManager {
 
     // Flag the instance for reloading when configs change
     garden.events.once("configChanged", (_payload) => {
-      if (!garden.needsReload) {
+      if (!garden.needsReload()) {
```
Comment (stefreak) [inline core/src/server/instance-manager.ts:239]: Is there possibly a linter rule that catches these kinds of bugs? @TimBeyer @eysi09

File: core/src/server/server.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (eysi09) [reply core/src/server/server.ts:895]: Haha on closer inspection this code doesn't make sense at all. I just noticed we were using `success` to print something that looked like an error but didn't look at the overall flow. My intention I was basically just to fix a typo. 

I'll update though.

**Review by stefreak — APPROVED** [review]

**Pull request merged**